src/dfm/data/load_datasets.py

```diff
@@ -0,0 +1,217 @@
+        years (Iterable[int]): A list of years to include.
+        probabilities (Optional[List[float]], optionak): Interleave probabilites of years.
+            Default to None, denoting equal probabilites.
+        kwargs: arguments to be passed forward to load_dataset
+    
+    Returns:
+        IterableDatasetDict: A datasets IterableDatasetDict
+    """
+    path_to_nat = Path(path_to_nat)
+
+    train_paths = []
+    for year in years:
+        train_path = path_to_nat / f"{year}_deduplicated_filtered.jsonl"
+        train_paths.append(str(train_path))
+    datasets = [load_dataset("json", data_files={"train": [path]*repeats}, streaming=True, split="train", **kwargs) 
+                for path in train_paths]
+    nat = interleave_datasets(datasets=datasets, probabilities=probabilities, seed=seed)
+    nat = IterableDatasetDict({"train": nat})
+    
+    _add_source = partial(__add_source, source = "nat")
+    nat = nat.map(_add_source)
+
+    if columns_to_keep:
+        nat = keep_columns(nat, columns_to_keep)
+    return nat
+
+def load_dcc_v1(
+    probabilities=[0.06, 0.06, 0.03, 0.85],
+    path_to_hopetwitter: Union[str, Path]=HOPETWITTER_PATH,
+    path_to_dagw: Union[str, Path]=DAGW_DFM_PATH,
+    path_to_danews: Union[str, Path]=DANEWS_PATH,
+    path_to_nat: Union[str, Path]=NAT_PATH
+):
+    columns_to_keep = ["text", "source"]
+    danews = load_danews(columns_to_keep=columns_to_keep, path_to_danews=path_to_danews, repeats=1_000)
+    dagw_dfm = load_dagw_dfm(columns_to_keep=columns_to_keep, path_to_dagw=path_to_dagw, repeats=1_000)
+    hopetwitter = load_hopetwitter(columns_to_keep = columns_to_keep, path_to_hopetwitter=path_to_hopetwitter, repeats=1_000)
+    nat = load_nat(columns_to_keep=columns_to_keep, path_to_nat=path_to_nat, repeats=100)
+    train = interleave_datasets([danews["train"], dagw_dfm["train"], hopetwitter["train"], nat["train"]], probabilities=probabilities)
+    # Note: NAT does not include a test, val set
+    # val = concatenate_datasets([danews["validation"], dagw_dfm["validation"], hopetwitter["validation"]])
+    # test = concatenate_datasets([danews["test"], dagw_dfm["test"], hopetwitter["test"]])
+
+    # as concatenate_datasets is not yet implemented for IterableDatasets:
+    dagw_test = Path(path_to_dagw) / "dagw_reddit_v1.0.0_test.jsonl"
+    dagw_val = Path(path_to_dagw) / "dagw_reddit_v1.0.0_val.jsonl"
+    danews_test = Path(path_to_danews) / "infomedia_2000-2021_v1.0.0_test.jsonl"
+    danews_val = Path(path_to_danews) / "infomedia_2000-2021_v1.0.0_val.jsonl"
+    twitter_test = Path(path_to_hopetwitter) / "twitter_da_stopwords_2019-01-01_2021-04-30_v1.0.0_test.jsonl"
+    twitter_val = Path(path_to_hopetwitter) / "twitter_da_stopwords_2019-01-01_2021-04-30_v1.0.0_val.jsonl"
+    data_files = {"test": [str(dagw_test), str(danews_test), str(twitter_test)], 
+                  "validation": [str(dagw_val), str(danews_val), str(twitter_val)]
+    }
```

Comment:
nat missing? 

Also, I'd prefer changing order to the order in the function arguments, and to separate each pair by newline.

Reply:
NAT does not have a test set. As we argued it was better to use an open source web scrape.